cheatsheets/Java_Security_Cheat_Sheet.md

```diff
@@ -0,0 +1,685 @@
+
+### Log Injection
+
+#### Symptom
+
+[Log Injection](https://owasp.org/www-community/attacks/Log_Injection) occurs when an application includes untrusted data in an application log message (e.g., an attacker can cause an additional log entry that looks like it came from a completely different user, if they can inject CRLF characters in the untrusted data). More information about this attack is available on the OWASP [Log Injection](https://owasp.org/www-community/attacks/Log_Injection) page.
+
+#### How to prevent
+
+To prevent an attacker from writing malicious content into the application log, apply defenses such as:
+
+- Filter the user input used to prevent injection of **C**arriage **R**eturn (CR) or **L**ine **F**eed (LF) characters.
+- Limit the size of the user input value used to create the log message.
+- Make sure [all XSS defenses](Cross_Site_Scripting_Prevention_Cheat_Sheet.md) are applied when viewing log files in a web browser.
+
+#### Example using Log4j2
+
+Configuration of a logging policy to roll on 10 files of 5MB each, and encode/limit the log message using the [Pattern *encode{}{CRLF}*](https://logging.apache.org/log4j/2.x/manual/layouts.html#PatternLayout\%7CLog4j2), introduced in [Log4j2 v2.10.0](https://mvnrepository.com/artifact/org.apache.logging.log4j/log4j-api), and the *-500m* message size limit.:
+
+``` xml
+<?xml version="1.0" encoding="UTF-8"?>
+<Configuration status="error" name="SecureLoggingPolicy">
+    <Appenders>
+        <RollingFile name="RollingFile" fileName="App.log" filePattern="App-%i.log" ignoreExceptions="false">
+            <PatternLayout>
+                <!-- Encode any CRLF chars in the message and limit its
+                     maximum size to 500 characters -->
+                <Pattern>%d{ISO8601} %-5p - %encode{ %.-500m }{CRLF}%n</Pattern>
+            </PatternLayout>
+            <Policies>
+                <SizeBasedTriggeringPolicy size="5MB"/>
+            </Policies>
+            <DefaultRolloverStrategy max="10"/>
+        </RollingFile>
+    </Appenders>
+    <Loggers>
+        <Root level="debug">
+            <AppenderRef ref="RollingFile"/>
+        </Root>
+    </Loggers>
+</Configuration>
+```
+
+Usage of the logger at code level:
+
+``` java
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+...
+// No special action needed because security actions are
+// performed at the logging policy level
+Logger logger = LogManager.getLogger(MyClass.class);
+logger.info(logMessage);
+...
+```
+
+#### Example using Logback with the OWASP Security Logging library
+
+Configuration of a logging policy to roll on 10 files of 5MB each, and encode/limit the log message using the [CRLFConverter](https://github.com/javabeanz/owasp-security-logging/wiki/Log-Forging), provided by the [OWASP Security Logging Project](https://owasp.org/www-project-security-logging/), and the *-500msg* message size limit:

```

Comment:
The link https://owasp.org/www-project-security-logging/ returns a 404 and needs fixed. (Looks like it is also wrong in the [README.md](https://github.com/augustd/owasp-security-logging/blob/master/README.md) for that project, where the readme references https://www.owasp.org/index.php/OWASP_Security_Logging_Project which redirects to the one I first mentioned, which is not found.

Reply:
I'm more concerned because it looks like the wiki page is hosed, so it's not really a specifically a CS problem. I looked for an alternative page but didn't find one. The best way to address it is probably to leave a GitHub issue on the https://github.com/javabeanz/owasp-security-logging repo. I've been swamped with other stuff lately, but if I get some spare cycles and remember to do so (which is the harder part), I will create one.